### PR TITLE
fix: Check if the user has another requests on the same day with any status

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Related Issues

- https://github.com/codescalers/CSHR/issues/378

### CODE REVIEW

**Review by zaelgohary — CHANGES_REQUESTED** [review]
When I create a request and have an approved request on the same day I get this.

image

image

**Comment by Mahmoud-Emad** [comment]
> When I create a request and have an approved request on the same day I get this.
> 
> image
> 
> image

This requested in https://github.com/codescalers/CSHR/issues/378#issuecomment-2309610511

**Comment by zaelgohary** [comment]
> When I create a request and have an approved request on the same day I get this.
> 
> image
> 
> image

@Mahmoud-Emad The message here is misleading as I'm having an approved request, not a pending one. can we rephrase it to `The user has another request with the same dates. Please review the previous request before submitting a new one.`

**Comment by Mahmoud-Emad** [comment]
@zaelgohary Please check

**Review by maayarosama — APPROVED** [review]

**Review by zaelgohary — APPROVED** [review]
